src/lib.rs

```diff
@@ -656,6 +658,52 @@ impl<L, R> Either<L, R> {
     }
 }
 
+#[cfg(feature = "never_type")]
+impl<L, R> Either<L, R>
+where
+    R: Into<!>,
+{
+    /// Returns the left value, but never panics
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(never_type)]
+    /// # use either::*;
+    /// let left: Either<_, !> = Left(3);
+    /// assert_eq!(left.into_left(), 3);
+    /// ```
+    pub fn into_left(self) -> L {
```

Comment:
On balance I don't think it's worth it to add usage of nightly feature for this.

- The feature gained is very minor - the gain is basically nothing
- Either is relatively widely used, stable, `1.x` release track and experimental features increase risk for breaking changes and increases maintenance required. The hurdle for using experimental features should be that they give a compelling advantage to users.

Reply:
The user can use `exhaustive_patterns` style code instead https://doc.rust-lang.org/std/primitive.never.html#infallible-errors

Reply:
Alright, makes sense, maybe once the feature hits stable it can be reconsidered again, but going over it, I do agree that it might not be worth it to add this requiring a nightly flag, I'll likely just make it an private extension trait in my crate, or just use `unwrap_*` which shouldn't panic nor emit any panic code in a release build (maybe even in a debug build too?).